packages/vite/src/node/utils.ts

```diff
@@ -1214,3 +1214,9 @@ const escapeRegexRE = /[-/\\^$*+?.()|[\]{}]/g
 export function escapeRegex(str: string): string {
   return str.replace(escapeRegexRE, '\\$&')
 }
+
+export function indexOfReg(str: string, reg: RegExp, pos: number = 0): number {
+  const matcher = str.slice(pos).match(reg)
+
+  return matcher?.index !== undefined ? matcher.index + pos : -1
+}
```

Comment:
Let's rename this function to `indexOfMatchInSlice` and move it to `importAnalysisBuild.ts`. Let's create shared utils when we hit the need for them. While refactoring, shared utilities are harder to work with.